src/pages/Settings/components/CreateCategoryForm.tsx

```diff
@@ -0,0 +1,100 @@
+/**
+ * Create category form component
+ *  - allow users to create a new category
+ *  - reusable for both expense and income category
+ *
+ */
+import React from 'react'
+import {useTranslation} from 'react-i18next'
+
+// Components
+import {Form, Input, Modal, Checkbox} from 'antd'
+import {FormComponentProps} from 'antd/es/form'
+
+// Utils
+import {enumToValues} from '../../../utils/utils'
+import {CategoryType} from '../../../models/Category'
+
+interface Props extends FormComponentProps {
+	form: any
+	visible: boolean
+	loading: boolean
+	title: string
+	handleSubmit: () => void
+	handleCancel: () => void
+}
+
+const CreateCategoryForm: React.FunctionComponent<Props> = React.forwardRef(
+	(props, ref) => {
+		const [t] = useTranslation(['common', 'settings'])
+		const {visible, loading, title, handleSubmit, handleCancel, form} = props
+		const {getFieldDecorator, setFieldsValue} = form
+
+		React.useImperativeHandle(ref, () => ({
+			form,
+		}))
+
+		const toggleCheckbox = e => {
```

Comment:
you can use radio button insteal of checkbox. Checkbox allows multiple choices while radio button does not